crates/bevy_render/src/world_sync.rs

```diff
@@ -0,0 +1,261 @@
+use bevy_app::Plugin;
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    observer::Trigger,
+    query::With,
+    reflect::ReflectComponent,
+    system::{Local, Query, ResMut, Resource, SystemState},
+    world::{Mut, OnAdd, OnRemove, World},
+};
+use bevy_hierarchy::DespawnRecursiveExt;
+use bevy_reflect::Reflect;
+use bevy_utils::tracing::warn;
+
+/// A Plugin that synchronizes entities with specific Components between the main world and the render world.
+///
+/// Bevy's renderer is architected independently from the main app.
+/// It operates in its own separate ECS World, so the renderer could run in parallel with the main world logic.
+/// This is called "Pipelined Rendering", see [`PipelinedRenderingPlugin`] for more information.
+///
+/// [`WorldSyncPlugin`] is the first thing that runs every frame and it maintains an entity-to-entity mapping
+/// between the main world and the render world.
+/// This is necessary for extraction, which copies over component data from the main to the render world.
+///
+/// ```text
+/// |--------------------------------------------------------------------|
+/// |      |         |          Main world loop                          |
+/// | sync | extract |---------------------------------------------------|
+/// |      |         |         Render world loop                         |
+//are/ |--------------------------------------------------------------------|
+/// ```
+///
+/// An example for synchronized main entities 1v1 and 18v1
+///
+/// ```text
+/// |---------------------------Main World----------------------------|
+/// |  Entity  |                    Component                         |
+/// |-----------------------------------------------------------------|
+/// | ID: 1v1  | PointLight | RenderEntity(ID: 3V1) | SyncRenderWorld |
+/// | ID: 18v1 | PointLight | RenderEntity(ID: 5V1) | SyncRenderWorld |
+/// |-----------------------------------------------------------------|
+///
+/// |----------Render World-----------|
+/// |  Entity  |       Component      |
+/// |---------------------------------|
+/// | ID: 3v1  | MainEntity(ID: 1V1)  |
+/// | ID: 5v1  | MainEntity(ID: 18V1) |
+/// |---------------------------------|
+///
+/// ```
+///
+/// Note that not this effectively establishes a link between the main world entity and the render world entity.
+/// Not every entity needs to be synchronized, however, only entities with [`SyncRenderWorld`] are synced.
```

Comment:
```suggestion
/// Not every entity needs to be synchronized, however; only entities with the [`SyncRenderWorld`] component are synced.
```

Or semicolon, or dash, idk english is hard.

Actually maybe just delete this whole sentence, as I asked to mention it in an earlier sentence/comment.